src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/WindowsForms/MyAppFileAccessor.cs

```diff
@@ -0,0 +1,181 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.WindowsForms;
+
+[Export(typeof(IMyAppXamlFileAccessor))]
+[AppliesTo(ProjectCapabilities.VB)]
+internal class MyAppFileAccessor : IMyAppXamlFileAccessor
+{
+    private readonly UnconfiguredProject _project;
+    private readonly IServiceProvider _serviceProvider;
+    private readonly IVsUIService<IVsRunningDocumentTable> _runningDocumentTable;
+    private readonly IProjectThreadingService _threadingService;
+
+    private readonly MyAppDocument _myappDocument;
+
+    private const string MySubMainProperty = "MySubMain";
+    private const string MainFormProperty = "MainForm";
+    private const string SingleInstanceProperty = "SingleInstance";
+    private const string ShutdownModeProperty = "ShutdownMode";
+    private const string EnableVisualStylesProperty = "EnableVisualStyles";
+    private const string AuthenticationModeProperty = "AuthenticationMode";
+    private const string SaveMySettingsOnExitProperty = "SaveMySettingsOnExit";
+    private const string HighDpiModeProperty = "HighDpiMode";
+    private const string SplashScreenProperty = "SplashScreen";
+    private const string MinimumSplashScreenDisplayTimeProperty = "MinimumSplashScreenDisplayTime";
+
+    [ImportingConstructor]
+    public MyAppFileAccessor(
+        UnconfiguredProject project,
+        IServiceProvider serviceProvider,
+        IVsUIService<IVsRunningDocumentTable> runningDocumnetTable,
+        IProjectThreadingService threadingService)
+    {
+        _project = project;
+        _serviceProvider = serviceProvider;
+        _runningDocumentTable = runningDocumnetTable;
+        _threadingService = threadingService;
+
+        _myappDocument = new MyAppDocument("Application.myapp", "filePath"); //TODO: get actual file name and file path.
+    }
+    public async Task<bool> GetMySubMainAsync()
+    {
+        await _threadingService.SwitchToUIThread();
```

Comment:
Do you need to switch to the UI thread in these methods?

If you do parse the file's contents on every read, you'll want to do that on the thread pool.

Reply:
Removed it in #8252 , thanks Drew!